runtime/POSIX/fd.c

```diff
@@ -425,6 +511,25 @@ int close(int fd) {
     return -1;
   }
 
+  // If the file was mmap-ed and not unmapped, we also want to check for 
+  // persistence.
```

Comment:
Why are we checking for persistence/adjusting ref counts in close? I was fairly certain with mmap the memory shouldn't be reclaimed when close() is invoked. See here: https://stackoverflow.com/questions/17490033/do-i-need-to-keep-a-file-open-after-calling-mmap-on-it

Specifically, "On the other hand, closing the file descriptor does not unmap the region."

Reply:
I just though it was a nice place to put a check. This is just a heuristic check that I thought might be useful.

Reply:
Re: our meeting earlier, I will remove this check.

Reply:
Actually, i need a variant of this check because a program can omit close and unmap calls and still exit. I will call the resolution at the termination of the state

Reply:
I remember now that's why I had this check